### PR TITLE
Remove SuppressWarning from msgs.hh to fix warnings

### PR DESCRIPTION
**Pull request by chapulina**
# 🦟 Bug fix

* Fixes a bunch of warnings introduced in https://github.com/ignitionrobotics/ign-msgs/pull/243
* ~~Requires https://github.com/ignitionrobotics/ign-cmake/pull/234~~

## Summary

Prevent the deprecated header from being included by anyone who includes `msgs.hh`.

This is technically not a soft tick, because someone relying on the suppression macros from `msgs.hh` would be broken, but I think there's probably zero people doing this.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸🔸

### CODE REVIEW

File: include/ignition/msgs/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,6 @@
 ign_install_all_headers(
   GENERATED_HEADERS
-    MessageTypes.hh)
+    MessageTypes.hh
+  INSTALL_BUT_DONT_INCLUDE
```
Comment (scpeters) [inline include/ignition/msgs/CMakeLists.txt:4]: I don't think that's a real parameter

* https://github.com/ignitionrobotics/ign-cmake/blob/ign-cmake2/cmake/IgnUtils.cmake#L601-L605

Comment (scpeters) [reply include/ignition/msgs/CMakeLists.txt:4]: I think you want to use `EXCLUDE_FILES` and then a separate install line

https://github.com/ignitionrobotics/sdformat/blob/sdf12/include/sdf/CMakeLists.txt#L1-L8

Comment (scpeters) [reply include/ignition/msgs/CMakeLists.txt:4]: oh, I missed https://github.com/ignitionrobotics/ign-cmake/pull/234

Comment (chapulina) [reply include/ignition/msgs/CMakeLists.txt]: Updated to your suggestion in [1c03d9e](https://github.com/ignitionrobotics/ign-msgs/pull/245/commits/1c03d9e91b26b924d1f178d5517e3938d7ec7c31)

**Review by caguero — COMMENTED** [review]
Should we add a note into `Migration.md`?

**Review by mjcarroll — APPROVED** [review]

**Comment by chapulina** [comment]
Migration note updated in [0fbde9f](https://github.com/ignitionrobotics/ign-msgs/pull/245/commits/0fbde9f682a30117a977ecb3a990e96b3edba319)

**Review by caguero — APPROVED** [review]

**Pull request merged**